src/layer.rs

```diff
@@ -80,6 +80,17 @@ impl LayerSet {
         self.layers.iter_mut().find(|l| &*l.name == name)
     }
 
+    /// Get a mutable reference to a layer, by name, or create it if it doesn't exist.
+    pub fn get_or_create_mut(&mut self, name: &str) -> &mut Layer {
+        if let Some(index) = self.layers.iter().position(|l| &*l.name == name) {
+            self.layers.get_mut(index).unwrap()
```

Comment:
maybe we can just do,
```suggestion
        if let Some(layer) = self.layers.iter_mut().find(|l| &*l.name == name) {
            Some(layer)
```

or else,

```rust
self.layers.iter_mut().find(|l| &*l.name == name)
    .or_else(make_the_layer)
```

But maybe this doesn't work because of that borrowchecker thing we discussed?

Reply:
I quickly tried

```rust
        self.layers.iter_mut().find(|l| &*l.name == name).or_else(move || {
            let layer = Layer::new(name.into(), None);
            self.layers.push(layer);
            self.layers.last_mut()
        }).unwrap()
```

but ran into the ownership issue again...